### DISCUSSION TITLE
Questions about API

### DISCUSSION THREAD

**Discussion by aur3l14no** [Future-Scholars/paperlib#259] [General]
I bulk import my whole library from zotero to paperlib. I noticed a lot of missing metadata so I select all of them and Ctrl + R. Boom. I got tons of 429.

I then realize that the metadata scraping takes place on your server, and the app communicates with it using API. So I have three questions here:

1. What is the rate limit of this API?
2. Does it scale? Will the server be flooded when the app has more users?
3. Do you consider open-sourcing the scraper and backend for this metadata API? Since this API plus a plugin can help solve the problems for people in other worlds. For example, Zotero. I believe many users are still stuck with zotero for whatever reasons. And you can really help them out.

**Comment by GeoffreyChen777** [comment]
Hi @aur3l14no


1. The rate limit is 50 / min.I guess you have a lot of papers in your library.
2. Currently, it's a single 2C server. My personal VPS. Although Paperplib is of an org called Future Scholar. There is only one main developer (me) and a friend with me. We don't have enough funding to make it scalable.
3. I know Zotero doesn't have a good metadata retrieval experience. That's the reason why I developed Paperlib. But currently, I cannot open source the API backend code and this API service should be only used for Paperlib for now due to the reason I mentioned in the anwser 2.

**Reply by aur3l14no** [reply]
I see! Thanks for your detailed replies.

**Comment by GeoffreyChen777** [comment]
@aur3l14no 

About your missing metadata, are they available in zotero but missing in Paperlib after you import them?

**Reply by aur3l14no** [reply]
No they weren't available in Zotero 😂
